### PR TITLE
Feature/Buttons

### LINKED ISSUE #48: Reusable Pill Button UI
Create a reusable component we'll call "Pill Button". It should:

* Have a title
* Option for title to be bold or regular
* Have a dark theme (i.e. purple with white text) and a light theme (i.e. white. with purple text and outline)
* Have an active and inactive state
* Not a fixed width (see examples)

**Design**
Check on the screenshot below
[Link to Design](https://www.figma.com/file/4cV84H1jyMo0rJEAl3HNnJ/FightPandemics-landing-page?node-id=6638%3A29348)

**VARIATIONS**:

Light theme & dark theme

Screen Shot 2020-05-13 at 9 40 09 PM

Inactive
Screen Shot 2020-05-13 at 9 44 31 PM

### PR DESCRIPTION
**Pull request by lucianoschillagi**
**What this PR does**:

### Checklist

- [] Compiler warnings resolved
- [] Linter warnings resolved
- [] User-facing strings in Localizable.strings file
- [] Unused code, print statements, and comments removed

**Which issue(s) this PR fixes**:

Resolves #
Create 4 pill reusable buttons (regular, bold, outline, inactive)

**Special notes for reviewers**:
For testing purposes, please put "ButtonPrototypeController" as initial View Controller

**Additional comments**:

### CODE REVIEW

**Comment by haidalinda** [comment]
hi @lucianoschillagi ! please make sure to fill out these sections of the PR template:

```
[] Compiler warnings resolved
[] Linter warnings resolved
[] User-facing strings in Localizable.strings file
[] Unused code, print statements, and comments removed
Which issue(s) this PR fixes:

Resolves # 
```

the checklist is intentionally there to remind you to do those things - even if its a given!

the `Resolves` section helps keep the Project Board in sync

File: FightPandemics/Features/Post/CreatePostViewController.swift [hunk 1]
```diff
@@ -67,6 +67,8 @@ class CreatePostViewController: UIViewController {
         thirdView.frame = CGRect(x: 0, y: 530, width: (UIScreen.main.bounds.width - 47), height: 300)
         self.view.addSubview(thirdView)
         thirdView.addSubview(createPostFooter)
+
+         print("HOLA")
```
Comment (haidalinda) [inline FightPandemics/Features/Post/CreatePostViewController.swift:69]: @lucianoschillagi the PR checklist asks not to leave print statements in

File: FightPandemics/Features/Profile/ProfileViewController.swift [hunk 2]
```diff
@@ -52,6 +52,8 @@ final class ProfileViewController: UIViewController {
                                                             style: .plain,
                                                             target: self,
                                                             action: #selector(logOut))
+
+        print("HOAA")
```
Comment (haidalinda) [inline FightPandemics/Features/Profile/ProfileViewController.swift:56]: remove print statements

File: FightPandemics/Features/Shared/Buttons/BoldButton.swift [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    // MARK: User Action Methods
+    @objc func boldButtonTapped() {
+        print("Bold Button Tapped!")
+    }
+
+    // MARK: Setup UI Methods
+    func setupBoldButton() {
+        setTitle("Post", for: .normal)
+        setTitleColor(.white, for: .normal)
+        titleLabel?.font = UIFont.systemFont(ofSize: 22.0, weight: .bold)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/BoldButton.swift:51]: see `Fonts.swift` for custom fonts

File: FightPandemics/Features/Shared/Buttons/BoldButton.swift [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    // MARK: User Action Methods
+    @objc func boldButtonTapped() {
+        print("Bold Button Tapped!")
+    }
+
+    // MARK: Setup UI Methods
+    func setupBoldButton() {
+        setTitle("Post", for: .normal)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/BoldButton.swift:49]: the checklist asks to Localized user-facing strings - but in this case, we do not want to bake in a title for the button; so this should be removed

File: FightPandemics/Features/Shared/Buttons/BoldButton.swift [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupBoldButton()
+    }
+
+    // MARK: User Action Methods
+    @objc func boldButtonTapped() {
+        print("Bold Button Tapped!")
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/BoldButton.swift:44]: remove print statement

File: FightPandemics/Features/Shared/Buttons/InactiveButton.swift [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    // MARK: Setup UI Methods
+    func setupInactiveButton() {
+        setTitle("Apply filters", for: .normal)
+        isEnabled = false
+        setTitleColor(.white, for: .normal)
+        backgroundColor = .blue
+        alpha = 0.5
+        titleLabel?.font = UIFont(name: "DMSans-Regular", size: 18)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/InactiveButton.swift:49]: use `Fonts.swift` enumeration

File: FightPandemics/Features/Shared/Buttons/InactiveButton.swift [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupInactiveButton()
+    }
+
+    // MARK: Setup UI Methods
+    func setupInactiveButton() {
+        setTitle("Apply filters", for: .normal)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/InactiveButton.swift:44]: remove default title

File: FightPandemics/Features/Shared/Buttons/OutlineButton.swift [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    // MARK: User Action Methods
+    @objc func outlineButtonTapped() {
+        print("Outline Button Tapped!")
+    }
+
+    // MARK: Setup UI Methods
+    func setupOutlineButton() {
+        setTitle("Close", for: .normal)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/OutlineButton.swift:49]: remove default title

File: FightPandemics/Features/Shared/Buttons/OutlineButton.swift [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+        setTitleColor(.blue, for: .normal)
+        backgroundColor = .clear
+        titleLabel?.font = UIFont(name: "AvenirNext-DemiBold", size: 18)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/OutlineButton.swift:52]: use `Fonts.swift` enumeration

File: FightPandemics/Features/Shared/Buttons/RegularButton.swift [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupRegularButton()
+    }
+
+    // MARK: User Action Methods
+    @objc func regularButtonTapped() {
+        print("Regular Button Tapped!")
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/RegularButton.swift:44]: remove print statment

File: FightPandemics/Features/Shared/Buttons/RegularButton.swift [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    // MARK: Setup UI Methods
+    func setupRegularButton() {
+//      setTitle("Setting the title here", for: .normal)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/RegularButton.swift:49]: the checklist asks to remove commented code

File: FightPandemics/Features/Shared/Buttons/OutlineButton.swift [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        setupOutlineButton()
+    }
+
+    // MARK: User Action Methods
+    @objc func outlineButtonTapped() {
+        print("Outline Button Tapped!")
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/OutlineButton.swift:44]: remove print statement

File: FightPandemics/Features/Shared/Buttons/RegularButton.swift [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+        setTitleColor(.white, for: .normal)
+        backgroundColor = .blue
+        titleLabel?.font = UIFont(name: "DMSans-Regular", size: 18)
```
Comment (haidalinda) [inline FightPandemics/Features/Shared/Buttons/RegularButton.swift:52]: use `Fonts.swift` enumeration

**Comment by haidalinda** [comment]
general comment - the button doesn't need to be 4 different buttons that duplicate style informatin. rather, 1 button (lets say `PillButton`) that is configured

**Review by haidalinda — CHANGES_REQUESTED** [review]

File: FightPandemics/Features/Shared/Buttons/RegularButton.swift (hunk 13 above)
Comment (lucianoschillagi) [reply FightPandemics/Features/Shared/Buttons/RegularButton.swift:52]: Ok for all your comments, I am going to work on them. Thanks.

**Comment by lucianoschillagi** [comment]
Hi @hkellaway , I did the changes. I have to do **another** PR and close this PR?

**Comment by haidalinda** [comment]
> Hi @hkellaway , I did the changes. I have to do **another** PR and close this PR?

hi @lucianoschillagi - response is here: https://fightpandemics.slack.com/archives/C012N43BHF1/p1589823981055800?thread_ts=1589585929.455600&cid=C012N43BHF1

**Comment by haidalinda** [comment]
gonna close in for replacement by this: https://github.com/FightPandemics/FightPandemics-iOS/pull/108

**Pull request closed without merging**